### PR TITLE
Issue #386

### LINKED ISSUE #386: Crypto functions are using Ethereum
**Is your feature request related to a problem? Please describe.**
Currently the staging branch is using Ethereum for all crypto functions. 

**Describe the solution you'd like**
I would like this library to be changed to Tendermint or Cosmos depending on the availability.

**(Optional) Include a BDD test that would satisfy the expected behaviour of the new feature**
BDD for all crypto functions is needed

### LINKED ISSUE #437: CircleCI is not running BDD tests
**Describe the bug**
The CI configuration is not running the BDD tests.

**To Reproduce**
Run any CI workflow for Pocket Core.

**Expected behavior**
BDD Tests must be run and output of the tests must be shown.

**Desktop (please complete the following information):**
CircleCI

### LINKED ISSUE #439: Fix AAT BDD test
**Describe the bug**
```
tests/bdd/aat/aat_test.go:22:37: expected '(', found '{'
tests/bdd/aat/aat_test.go:23:10: expected type, found "should return missing version error"
tests/bdd/aat/aat_test.go:24:5: expected 'IDENT', found '}'
tests/bdd/aat/aat_test.go:26:14: expected type, found "Missing message"
tests/bdd/aat/aat_test.go:27:9: missing ',' in parameter list
tests/bdd/aat/aat_test.go:28:5: expected 'IDENT', found '}'
tests/bdd/aat/aat_test.go:30:14: expected type, found "Missing appPubKey"
tests/bdd/aat/aat_test.go:31:9: missing ',' in parameter list
tests/bdd/aat/aat_test.go:32:5: expected 'IDENT', found '}'
tests/bdd/aat/aat_test.go:34:14: expected type, found "Missing signature"
tests/bdd/aat/aat_test.go:35:9: missing ',' in parameter list
```

**To Reproduce**
`go test ./tests/bdd/aat/aat_test.go`

**Expected behavior**
Tests to run succesfully.

### PR DESCRIPTION
**Pull request by luyzdeleon**
Closes #439 
Closes #437 
Closes #386

### CODE REVIEW

File: crypto/hash.go [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
 import (
-	"crypto"
+	"golang.org/x/crypto/sha3"
 )
 
-// TODO convert to standard crypto library
-func Hash(input []byte) []byte {
-	hasher := crypto.SHA3_256.New()
-	hasher.Write(input)
+// Converts []byte to SHA3-256 hashed []byte
```
Comment (andrewnguyen22) [inline crypto/hash.go:7]: Would be nice to have hex wrapper functions in crypto package as well

File: crypto/keys.go [hunk 2]
```diff
@@ -1,24 +1,111 @@
...
+	ecdsaPubKey ecdsa.PublicKey
+}
+
+// wrapper around go-secp256k1 private key
+type PrivateKey struct {
+	ecdsaPrivKey *ecdsa.PrivateKey
+}
+
+// Private Key functions
+// TODO: Create a function to import private key from []byte or hex dump.
```
Comment (andrewnguyen22) [inline crypto/keys.go:21]: I don't think you satisfied this TODO

File: crypto/keys.go [hunk 3]
```diff
@@ -1,24 +1,111 @@
...
+		return nil
+	}
+	return elliptic.Marshal(S256(), pubKey.ecdsaPubKey.X, pubKey.ecdsaPubKey.Y)
+}
 
-func NewKeypair() (privateKey PrivateKey, publicKey PublicKey) {
-	privateKey = NewPrivateKey()
-	publicKey = privateKey.PubKey()
-	return
+// Creates a PublicKey instance from bytes
```
Comment (andrewnguyen22) [inline crypto/keys.go:63]: I could use some usage comments here

File: crypto/keys.go [hunk 4]
```diff
@@ -1,24 +1,111 @@
...
+	pubKey = privKey.GetPublicKey()
+
+	return privKey, pubKey, nil
 }
 
-// todo need function to convert hex string or bytes into public key
+// Signature Functions
+
+// "GetPublicKeyFromSignature" confirms the public key from the signature.
+func GetPublicKeyBytesFromSignature(messageHash, signature []byte) ([]byte, error) {
```
Comment (andrewnguyen22) [inline crypto/keys.go:90]: Comment and function name do not match

File: crypto/keys.go [hunk 5]
```diff
@@ -1,24 +1,111 @@
...
+	// The signature should be in [R || S] format. !!! Remove V (last byte)
+	return secp256k1.VerifySignature(pubKey.Bytes(), messageHash, signature[:len(signature)-1])
+}
+
+// "VerifySignature" verifies the signature with the public key bytes.
+func VerifySignatureWithPubKeyBytes(pubKeyBytes []byte, messageHash []byte, signature []byte) bool {
+	// VerifySignatureWithPubKeyBytes checks that the given pubkey bytes created signature over message.
+	// The signature should be in [R || S] format. !!! Remove V (last byte)
+	return secp256k1.VerifySignature(pubKeyBytes, messageHash, signature[:len(signature)-1])
+}
```
Comment (andrewnguyen22) [inline crypto/keys.go:110]: How do we convert public key bytes into compatible hex string? hex.EncodeToString()? is it that easy?

File: error/error.go [hunk 6]
```diff
@@ -0,0 +1,67 @@
+package error
```
Comment (andrewnguyen22) [inline error/error.go]: This is the legacy error package that was deprecated in one of the first PRs. You should keep your error.go file in the crypto package for a better design and remove this file. See `Staging` for clarity on how errors are supposed to be organized

**Review by andrewnguyen22 — CHANGES_REQUESTED** [review]
Run `go fmt ./...` when you are finished cause there was a few formatting things.

File: crypto/keys.go (hunk 3 above)
Comment (andrewnguyen22) [reply crypto/keys.go:63]: Not required, but would be helpful!

File: crypto/hash.go (hunk 1 above)
Comment (luyzdeleon) [reply crypto/hash.go:7]: https://github.com/pokt-network/pocket-core/issues/444

File: crypto/keys.go (hunk 2 above)
Comment (luyzdeleon) [reply crypto/keys.go:21]: Yes, I couldn't find a suitable solution, so I created the following issue: https://github.com/pokt-network/pocket-core/issues/445

File: crypto/keys.go (hunk 5 above)
Comment (luyzdeleon) [reply crypto/keys.go:110]: Covered by this issue: https://github.com/pokt-network/pocket-core/issues/444

**Review by andrewnguyen22 — APPROVED** [review]

**Pull request merged**